convert2rhel/unit_tests/cert_test.py

```diff
@@ -16,77 +16,108 @@
 import os
 import shutil
-import unittest
 
 import pytest
-import six
 
-from convert2rhel import unit_tests
+from convert2rhel import cert, unit_tests, utils
+from convert2rhel.systeminfo import system_info
+from convert2rhel.toolopts import tool_opts
 
 
-six.add_move(six.MovedModule("mock", "mock", "unittest.mock"))
+# Directory with all the tool data
+BASE_DATA_DIR = os.path.realpath(os.path.join(os.path.dirname(__file__), "../data/"))
 
-from convert2rhel import cert, utils
-from convert2rhel.systeminfo import system_info
-from convert2rhel.toolopts import tool_opts
 
+@pytest.fixture
+def cert_dir(monkeypatch, request):
+    if request.param["data_dir"] is None:
+        data_dir = unit_tests.NONEXISTING_DIR
+        rhel_cert_dir = os.path.join(data_dir, "rhel-certs")
+    else:
+        data_dir = request.param["data_dir"]
+        rhel_cert_dir = os.path.join(data_dir, "rhel-certs")
 
-class TestCert(unittest.TestCase):
-
-    # Certificates for all the supported RHEL variants
-    certs = {
-        "x86_64": {"6": "69.pem", "7": "69.pem", "8": "479.pem"},
-        "ppc64": {
-            "7": "74.pem",
-        },
-    }
-
-    # Directory with all the tool data
-    base_data_dir = os.path.realpath(os.path.join(os.path.dirname(__file__), "../data/"))
-
-    @unit_tests.mock(utils, "DATA_DIR", unit_tests.TMP_DIR)
-    def test_get_cert_path(self):
-        # Check there are certificates for all the supported RHEL variants
-        for arch, rhel_versions in self.certs.items():
-            for rhel_version, pem in rhel_versions.items():
-                utils.DATA_DIR = os.path.join(self.base_data_dir, rhel_version, arch)
-                system_cert = cert.SystemCert()
-                cert_path = system_cert._source_cert_path
-                self.assertEqual(cert_path, "{0}/rhel-certs/{1}".format(utils.DATA_DIR, pem))
-
-    @unit_tests.mock(cert, "loggerinst", unit_tests.GetLoggerMocked())
-    @unit_tests.mock(utils, "DATA_DIR", unit_tests.TMP_DIR)
-    @unit_tests.mock(system_info, "arch", "arch")
-    def test_get_cert_path_missing_cert(self):
         # Create temporary directory that has no certificate
-        cert_dir = os.path.join(utils.DATA_DIR, "rhel-certs", system_info.arch)
+        cert_dir = os.path.join(rhel_cert_dir, request.param["arch"])
         utils.mkdir_p(cert_dir)
-        # Check response for the non-existing certificate in the temporary dir
-        self.assertRaises(SystemExit, cert.SystemCert._get_cert)
-        self.assertEqual(len(cert.loggerinst.critical_msgs), 1)
-        # Remove the temporary directory tree
+
+    monkeypatch.setattr(utils, "DATA_DIR", data_dir)
+    monkeypatch.setattr(system_info, "arch", request.param["arch"])
+
+    yield rhel_cert_dir
+
+    # Remove the temporary directory tree if we created it earlier
+    if request.param["data_dir"] is not None:
         shutil.rmtree(os.path.join(utils.DATA_DIR, "rhel-certs"))
```

Comment:
I would change this to `data_dir` instead of using the constant in `utils`, because if someone changes this fixture in some way, or we change the constant name, this fixture would start failing.

```suggestion
        shutil.rmtree(data_dir)
```

Reply:
@abadger, I have pushed the fix here.